### DISCUSSION TITLE
Python

### DISCUSSION THREAD

**Discussion by sree9578** [LibreDWG/libredwg#402] [Q&A]
Hi @rurban 

 after installing how do i configure this package for python on windows?
i built the package ...but how can i use it as python package ..how to import libredwg as said in examples..

```
"C:\Python310\DLLs\LibreDWG.py", line 15, in <module>
    import _LibreDWG
ModuleNotFoundError: No module named '_LibreDWG'
```
Thanks

**Comment by nklsla** [comment]
I'm having the same problem on my windows machine. Is it possible to import this as an library without building it? In other words, can I use the pre-built download package for windows and some how configure it to run in python?

**Comment by rurban** [comment]
The prebuilt mingw packages do include the python bindings, 
but you need to have the very same python version as mingw.
And you need to install them by yourself into your python site-packages

See build-aux\msys2.bat and build-aux\appveyor-deploy.bat
=> lib/python3.10/site-packages/_LibreDWG-0.dll

**Reply by nklsla** [reply]
Thanks I'll try that!

**Comment by nklsla** [comment]
One workaround that works for me since my goal is to extract text from drawings.
Either convert DWG->DXF and search the DXF-file for text (a DXF file is just a glorified .txt-file) or search directly in the file with regex.
I couldn't use the executables directly due to my companys IT-security but calling them from python worked. See example below:

```
import subprocess
import glob

paths = glob.glob(r"C:\path\to\dwgfiles\*.dwg")
for path in paths:
    subprocess.call(
        # Case 1
        # Find dates with regex in format NNN-NN-NN
        # [r"C:\projekt\libredwg\dwggrep.exe", r"\d{4}-\d{2}-\d{2}", path]
        
        # Case 2
        # Convert to DXF
        [r"C:\projekt\libredwg\dwg2dxf.exe", "-v0", "-y", path]  
    )
```

**Comment by BoTan-sudo** [comment]
I'm having the same problem on my windows machine today, maybe python on windows perfer to _LibreDWG.pyd than _LibreDWG.dll
